tuna/connection.py

```diff
@@ -43,37 +45,47 @@
 class Connection():
   """Connection class defined an ssh or ftp client connection. Instantiated by the machine class"""
 
+  # pylint: disable=too-many-instance-attributes
   #pylint: disable=no-member
-
-  def __init__(self, **kwargs):
+  def __init__(self, **kwargs: dict) -> None:
     #pylint
-    self.logger = None
-    self.local_machine = False
-    self.subp = None  # Holds the subprocess obj for local_machine
-    self.out_channel = None  # Holds the out channel for remote connection
+    self.logger: Any = None
+    self.local_machine: bool = False
+    self.subp: Any = None  # Holds the subprocess obj for local_machine
+    self.out_channel: Any = None  # Holds the out channel for remote connection
+
+    #initialize the class member variables
+    self.id = 0  # pylint: disable=invalid-name
+    self.hostname = ' '
+    self.port = 0
+    self.user = None
+    self.password = None
+
+    self.ssh: Any = None
 
-    allowed_keys = set([
+    allowed_keys: Set[str] = set([
         'id', 'hostname', 'user', 'password', 'port', 'local_machine',
         'chk_abort_file', 'logger'
     ])
+
     self.__dict__.update((key, None) for key in allowed_keys)
     self.__dict__.update(
         (key, value) for key, value in kwargs.items() if key in allowed_keys)
 
-    self.ssh = None
-
     if self.logger is None:
       self.logger = setup_logger('Connection')
 
     self.inst_bins = {'which': True, 'cd': True}
-    self.connect(self.chk_abort_file)
+    self.connect(chk_abort_file)
```

Comment:
self shouldn't be removed here

Reply:
added self to check abort call

Reply:
self.chk_abort_file = chk_abort_file added as object attribute set. In init routine